### PR TITLE
Improve DH integration tests and reorganize test suite

### LINKED ISSUE #663: Improve the DH integration tests
I'd like to see the door hanger test improved to include:
* tests reorganized into functional suites that match TestRail (scoped to include the non-DH tests)
* logging into FxA
* when creating the entry, putting content into each of the input fields
* instead of verifying the entry fields are blank, update the test to verify the name is correct

A future task will be to improve the test to validate the remainder of the entries data data fields are correct.

### PR DESCRIPTION
**Pull request by m8ttyB**
Resolves #663 

* A slight update to the existing door hanger test.
* Updates `new_entry()` to `create_new_entry()` which can take args.
* Begins to move away from the `munged_class_name()` helper and uses CSS selectors.
* Moves the remaining test cases into suites that match the canonical suites in TestRail.

### CODE REVIEW

File: test/integration/pages/home.py [hunk 1]
```diff
@@ -50,11 +48,18 @@ def lockie(self):
...
+        """Create and save a new entry."""
         current_entries = len(self.entries)
         self.find_element(*self._new_entry_locator).click()
-        self.find_element(*self._save_entry_locator).click()
+        entry = self.entries[0].click()
+        entry.set_site_name(site_name)
+        entry.set_website(url)
+        entry.set_username(username)
+        entry.set_password(password)
+        entry.set_note(note)
```
Comment (jrbenny35) [inline test/integration/pages/home.py]: Is there a reason we would only set 1 of these?

Comment (m8ttyB) [reply test/integration/pages/home.py]: Do mean can a user have multiple notes per entry? At this time, no.

Comment (jrbenny35) [reply test/integration/pages/home.py]: No, I am asking if we would only use ```set_username``` for example, or would it be all of them every time?

Comment (m8ttyB) [reply test/integration/pages/home.py]: I follow now, thanks. I think for this first iteration I was thinking yes. The next round of work I want to move things around a bit more and could see this method returning an Entry object and the individual test touches the setter methods. Do you see this as blocking or can this happen in a separate pr?

Comment (jrbenny35) [reply test/integration/pages/home.py]: Not it's not blocking. I think we could iterate on this though. Maybe we don't need so many ```set_``` methods.

Comment (m8ttyB) [reply test/integration/pages/home.py]: Agreed. I felt unclean doing so.

**Comment by devinreams** [comment]
Marking this as a candidate for 0.1.9 release.

This assumes a 👍 review from @jimporter within the day or so.

**Pull request merged**